README.md

```diff
@@ -196,6 +196,22 @@ defmodule MyApp.ElasticsearchMock do
 end
 ```
 
+#### Elasticsearch.API.AWS
+
+As AWS does not provide credentials' based http authentication, you can use the `Elasticsearch.API.AWS` module if you want to use AWS Elasticsearch Service with AWS Signature V4 signed HTTP connections.
+
+To use this, just add to your config:
+
+```elixir
+# config/prod.exs
+config :my_app, MyApp.ElasticsearchCluster,
+  api: Elasticsearch.API.AWS,
+  aws_region: "us-east-1",
+  aws_service: "es",
+  aws_access_key_id: "aws_access_key_id",
+  aws_secret_access_key: "aws_secret_access_key"
```

Comment:
Could we pass these `aws_` variables in `default_options` instead? These options are passed straight through to the API module, and it won't require us to add API-specific keys to the cluster config.

Reply:
I thought about it but then it felt weird that we're treating credentials in a different way and also those variables aren't mandatory unless the API.AWS module is specified. Do you think it is paramount?

Reply:
In `v2.0.0` (which is in development) options like this would be configured like this:

```elixir
config :my_app, MyApp.Cluster,
  api: Elasticsearch.API.HTTPoison.AWS,
  api_opts: [
    aws_region: "us-east-1",
    aws_service: "es",
    aws_access_key_id: "aws_access_key_id",
    aws_secret_access_key: "aws_secret_access_key"
  ]
```

The `api_opts` allow you to pass any arbitrary list of configuration that the API adapter might need, and allows us to support a wide range of adapters without adding top-level config options. 

If we pass these options as part of `default_options` for now, it will be more similar to how things will work in 2.0.0, so I would prefer that.

Reply:
I applied the suggested changes with one minor detail, instead of creating aws_ variables I created a nested List under aws in default_options so that it is cleaner to access and clean from further propagating all credentials on the request.